src/pages/postgraphile/smart-tags.md

```diff
@@ -394,6 +408,35 @@ $$ language SQL STABLE;
 comment on function user_object_field() is E'@sortable';
 ```
 
+Note that this will only work with a single composite type. If your computed
+function returns a `SETOF` composite type, you will need to use an `ARRAY` and
+the
+[connection-filter plugin](https://github.com/graphile-contrib/postgraphile-plugin-connection-filter).
+
+```sql
+-- non scalar setof function
+CREATE OR REPLACE FUNCTION user_objects(user user) RETURNS SETOF object AS $$
+SELECT * FROM object where owner_id = user.id; -- one user, many objects
+$$ language SQL STABLE;
+
+-- wrapper. Note the () for notation. Failing to use them will throw an error
+CREATE OR REPLACE FUNCTION user_object_fields(user user) RETURNS varchar[] AS $$
+SELECT ARRAY_AGG((user_objects(user)).field);
+$$ language SQL STABLE;
+
+-- don't forget the comments...
+comment on function user_object_fields() is E'@sortable';
+```
+
+Now you can use the array as a _filter by_, for example:
+
+```gql
+{
+  # get all users who own an object with field == 'foo'
+  allUsers(filter: {object_fields: contains: 'foo'}) { ... }
+}
+```
```

Comment:
This section belongs in the postgraphile-plugin-connection-filter project; I typically don't recommend the addition of arbitrary filters to a GraphQL API, so I'd rather not have this example so prominently in our docs. Happy to link out to it though :+1:

Reply:
Would you be ok to leave the example with a big warning saying something like "this is not recommend, but in case you want to do it..."? 

As a user, I prefer to see how things are done and understand the cost of doing it than to hide it, but I also understand your opinion on not showing it.

Collateral question, what do you recommend? A custom query with the filter parameter? We could also show an example on the recommended way to do it

Reply:
Yeah I'd probably use a custom query, or possibly https://www.graphile.org/postgraphile/make-add-pg-table-condition-plugin/